shared/teams/confirm-modals/confirm-kick-out.tsx

```diff
@@ -0,0 +1,156 @@
+import * as React from 'react'
+import * as Kb from '../../common-adapters'
+import * as Container from '../../util/container'
+import * as Types from '../../constants/types/teams'
+import * as Styles from '../../styles'
+import * as Constants from '../../constants/teams'
+import * as TeamsGen from '../../actions/teams-gen'
+import {memoize} from '../../util/memoize'
+
+type Props = Container.RouteProps<{members: string[]; teamID: Types.TeamID}>
+
+const membersStr = (members: string[]): string => {
+  switch (members.length) {
+    case 0:
+      return 'nobody'
+    case 1:
+      return members[0]
+    case 2:
+      return `${members[0]} and ${members[1]}`
+    case 3:
+      return `${members[0]}, ${members[1]} and ${members[2]}`
+    default:
+      return `${members[0]}, ${members[1]}, and ${members.length - 2} others`
+  }
+}
+
+const getSubteamNames = memoize((state: Container.TypedState, teamID: Types.TeamID): [
+  string[],
+  Types.TeamID[]
+] => {
+  const subteamIDs = [...Constants.getTeamDetails(state, teamID).subteams]
+  return [subteamIDs.map(id => Constants.getTeamMeta(state, id).teamname), subteamIDs]
+})
+
+const ConfirmKickOut = (props: Props) => {
+  const members = Container.getRouteProps(props, 'members', [])
+  const teamID = Container.getRouteProps(props, 'teamID', Types.noTeamID)
+  const [subteamsToo, setSubteamsToo] = React.useState(false)
+
+  const [subteams, subteamIDs] = Container.useSelector(state => getSubteamNames(state, teamID))
+  const teamname = Container.useSelector(state => Constants.getTeamMeta(state, teamID).teamname)
+  const waitingKeys = ([] as string[]).concat.apply(
+    [],
+    members.map(member => subteamIDs.map(subteamID => Constants.removeMemberWaitingKey(subteamID, member)))
+  )
+  const waiting = Container.useAnyWaiting(...waitingKeys)
+
+  const dispatch = Container.useDispatch()
+  const nav = Container.useSafeNavigation()
+  const onCancel = React.useCallback(() => dispatch(nav.safeNavigateUpPayload()), [dispatch, nav])
```

Comment:
why `useCallback` here?

Reply:
Because I'm passing it as a `dep` to the `useEffect` below